extensions/amp-addthis/0.1/addthis-utils/cuid.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const RE_CUID = /^[0-9a-f]{16}$/;
+const MAX_HEX = 0xffffffff;
+const CUID_SESSION_TIME = Date.now();
+
+/**
+ * Get the date from the CUID (first 8 hex digits are the time from epoch in seconds).
+ * @param {string} cuid
+ * @returns {Date}
+ */
+const getDateFromCuid = cuid => {
+  let date = new Date();
+  try {
+    date = (new Date((parseInt(cuid.substr(0, 8), 16) * 1000)));
+  } catch (e) {}
+  return date;
+};
+
+/**
+ * Check if the CUID is in the future (allowing for up to one day of jitter).
+ * @param {string} cuid
+ * @returns {boolean}
+ */
+const isCuidInFuture = cuid => {
+  const date = getDateFromCuid(cuid);
+  date.setDate(date.getDate() - 1);
+  return isDateInFuture(date);
+};
+
+export const isDateInFuture = date => {
+  const now = new Date();
+  if (date.getFullYear() < now.getFullYear()) {
+    return false;
+  }
+  const yearIsLater = date.getFullYear() > now.getFullYear();
+  const yearIsSame = date.getFullYear() === now.getFullYear();
+  const monthIsLater = date.getMonth() > now.getMonth();
+  const monthIsSame = date.getMonth() === now.getMonth();
+  const dateIsLater = date.getDay() > now.getDay();
```

Comment:
@jmrog @honeybadgerdontcare @aghassemi This line is incorrect, since it is comparing the day of the week, and not the date. As written, this logic will break every Saturday.

See https://travis-ci.org/ampproject/amphtml/jobs/372268665#L1472-L1476

It should be replaced with `date.getDate() > now.getDate()`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDay
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDate

Reply:
LOL. Please file a bug and lets get the test disabled.

Reply:
Sent out a fix instead. #14960